internal/pyxis/submit_test.go

```diff
@@ -101,6 +107,40 @@ var _ = Describe("Pyxis Submit", func() {
 				Expect(certResults).To(BeNil())
 			})
 		})
+		Context(" and the --offline flag is set", func() {
+			JustBeforeEach(func() {
+				viper.Set("offline", true)
+
+				tmpDir, err := os.MkdirTemp("", "preflight-submit-test-*")
+				Expect(err).ToNot(HaveOccurred())
+				DeferCleanup(os.RemoveAll, tmpDir)
+
+				// creating test files on in the tmpDir so the tar function has files to tar
+				f1, err := os.Create(filepath.Join(tmpDir, "test-file-1.json"))
+				Expect(err).ToNot(HaveOccurred())
+				defer f1.Close()
+
+				f2, err := os.Create(filepath.Join(tmpDir, "test-file-1.json"))
+				Expect(err).ToNot(HaveOccurred())
+				defer f2.Close()
+
+				aw, err := artifacts.NewFilesystemWriter(artifacts.WithDirectory(tmpDir))
+				Expect(err).ToNot(HaveOccurred())
+
+				artifactsContext = artifacts.ContextWithWriter(context.Background(), aw)
+			})
+			JustAfterEach(func() {
+				viper.Set("offline", false)
+			})
+			It("should write a tar to disk", func() {
+				certResults, err := pyxisClient.SubmitResults(artifactsContext, &certInput)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(certResults).ToNot(BeNil())
+				Expect(certResults.CertProject).ToNot(BeNil())
+				Expect(certResults.CertImage).ToNot(BeNil())
+				Expect(certResults.TestResults).ToNot(BeNil())
+			})
+		})
```

Comment:
Negative testing with a bad writer perhaps?

Reply:
I've added one negative test to not add an `ArtifactsWriter` to the context, which is the only error that can be tested in the main flow. 

I'll add `artifactsTar` method level testing in it's own `Context()` block, I don't think it makes sense at this location and should live on it's own.

Reply:
More tests have been added. I was not able to cover the `Irregular` file or `WriteHeader` errors, since whatever I passed in, either failed in the test setup, or failed before either of these method calls. 

If you feel covering those are important, let me know your ideas to cover them with proper 'fake' files and 'permissions' that fall through the other code, but fails in either case.

Reply:
More tests have been added since this code moved out of the submit flow.